### ISSUE TITLE
Docker-compose add-host option

### ISSUE THREAD

**Issue by nlapshin** [docker/compose#6355]
There is an option 'add-host' in docker, it allows to add hostname exactly during container assembly. Is there an implementation of this option for docker-compose?

I've seen the this question was discussed in issue https://github.com/docker/compose/issues/5282/ But it is closed and there is no practical sample that works.

**Comment by shin-** [comment]
https://docs.docker.com/compose/compose-file/#extra_hosts

**Comment by nlapshin** [comment]
> https://docs.docker.com/compose/compose-file/#extra_hosts

Thanks for answer. I have tried this option. But, if I understand correctly, 'extra_hosts' added hostname after container run.

In my case I need to use hostname during container assembly. I use docker-compose version 3.7.

This is example for docker.
```sh
docker build --add-host=docker:10.180.0.1
```

But, this example for docker-compose(with 'extra_hosts' option in the config) 
```sh
docker-compose build
```

These examples don't work identically. In the case of the docker example, I can use additional hostname, but, in the case of the docker-compose exampe, I cannot use it.

**Comment by shin-** [comment]
[You can use `extra_hosts` in the `build` section of your config definition, but only with v2 Compose files](https://docs.docker.com/compose/compose-file/compose-file-v2/#extra_hosts). 

If you're set on using v3, you'll want to follow this issue instead: https://github.com/docker/cli/issues/1293

**Comment by nlapshin** [comment]
Thanks a lot!

**Comment by burka** [comment]
For anyone searching like me: It's solved in Version 3.9 https://github.com/docker/cli/pull/2517 - you can add `extra_hosts:` in `build:` now.
